From e7d02b04dcfb65d1873b81bd3281442ac5ddec1d Mon Sep 17 00:00:00 2001
From: Henrik Gramner <hengar-6@student.ltu.se>
Date: Wed, 11 Jan 2012 17:40:34 +0000
Subject: [PATCH 08/67] fft: init functions with INIT_XMM/YMM.

This is required to handle clobbering of XMM registers on Win64
correctly. Fixes FFT and all tests depending on FFT on Win64.

Signed-off-by: Ronald S. Bultje <rsbultje@gmail.com>
Signed-off-by: Janne Grunau <janne-libav@jannau.net>
---
 libavcodec/x86/fft_mmx.asm |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/libavcodec/x86/fft_mmx.asm b/libavcodec/x86/fft_mmx.asm
index 27276a1..d6d07b8 100644
--- a/libavcodec/x86/fft_mmx.asm
+++ b/libavcodec/x86/fft_mmx.asm
@@ -639,11 +639,14 @@ cglobal fft_dispatch%3%2, 2,5,8, z, nbits
 %endmacro ; DECL_FFT
 
 %ifdef HAVE_AVX
+INIT_YMM
 DECL_FFT 6, _avx
 DECL_FFT 6, _avx, _interleave
 %endif
+INIT_XMM
 DECL_FFT 5, _sse
 DECL_FFT 5, _sse, _interleave
+INIT_MMX
 DECL_FFT 4, _3dn
 DECL_FFT 4, _3dn, _interleave
 DECL_FFT 4, _3dn2
-- 
1.7.5.4

